docs/source/migration/4.0.mdx

```diff
@@ -5,31 +5,82 @@ description: From 3.x
 
 <Note>
 
-**Version 4 is currently under development and this page itself is a work in progress.**
+**Version 4.0 is currently under development and this page itself is a work in progress.**
 
 </Note>
 
 Apollo Kotlin 3.0 was a major rewrite of Apollo in Kotlin multiplatform.
 
 Apollo Kotlin 4.0 focuses on tooling, stability and fixing some API regrets that came with 3.x.
 
-Because most of the common APIs stayed the same, we [kept the package name unchanged](https://github.com/apollographql/apollo-kotlin/issues/4710). Apollo Kotlin 4.0 removes some deprecated symbols. We strongly recommend removing deprecated usages before migrating to 4.0.  
+While most of the common APIs stayed the same, 4.0 isn't binary compatible with 3.x so we [changed the package name](https://github.com/apollographql/apollo-kotlin/issues/4710) to `com.apollographql.apollo`. 
+
+Apollo Kotlin 4.0 removes some deprecated symbols. We strongly recommend removing deprecated usages before migrating to 4.0.  
 
 If you are using a library that depends on Apollo Kotlin transitively, you need it to update to 4.x before you can update your own application to 4.0.
 
 ## Automatic migration using the Android Studio plugin
 
-Apollo Kotlin 4 ships with a companion [Android Studio plugin](../testing/android-studio-plugin#migration-helpers) that automates most of the migration. 
+Apollo Kotlin 4.0 ships with a companion [Android Studio plugin](../testing/android-studio-plugin#migration-helpers) that automates most of the migration. 
 
-It automates most of the API changes but cannot deal with behaviour changes like error handling. 
+It automates most of the API changes but cannot deal with behavior changes like error handling. 
 
 We recommend using the plugin to automate the repetitive tasks but still go through this document for the details. 
 
+## Package name / group id / plugin id
```

Comment:
Maybe this section should be merged with the "Maven artifacts reorganization" below.

Reply:
I'd say it's fine to have it prominently at the beginning. "Maven artifacts reorganization" is a lot more advanced APIs so I'm OK to have them in 2 different sections as it's probably 2 different audiences.

Nit: use the same order as what people are going to use when updating:

```suggestion
## Group id / plugin id / package name
```